matchesWhereQuery.js

```diff
@@ -3,45 +3,69 @@ var _ = require( 'underscore' );
+			if( ! queryAttribute.value ) throw new Error( 'Value must be supplied for comparator queries' );
 
-			switch( thisKeyQuery.comparator ) {
+			switch( queryAttribute.comparator ) {
 			case 'doesNotEqual':
-				if( thisObjectValue === thisKeyQuery.value ) return false;
+				if( objectAttribute === queryAttribute.value ) return false;
 				break;
 			case 'isGreaterThan':
-				if( thisObjectValue <= thisKeyQuery.value ) return false;
+				if( objectAttribute <= queryAttribute.value ) return false;
 				break;
 			case 'isGreaterThanOrEqualTo':
-				if( thisObjectValue < thisKeyQuery.value ) return false;
+				if( objectAttribute < queryAttribute.value ) return false;
 				break;
 			case 'isLessThan':
-				if( thisObjectValue >= thisKeyQuery.value ) return false;
+				if( objectAttribute >= queryAttribute.value ) return false;
 				break;
 			case 'isLessThanOrEqualTo':
-				if( thisObjectValue > thisKeyQuery.value ) return false;
+				if( objectAttribute > queryAttribute.value ) return false;
 				break;
 			case 'isBetween':
-				if( ! _.isArray( thisKeyQuery.value ) || thisKeyQuery.value.length !== 2 ) {
+				if( ! _.isArray( queryAttribute.value ) || queryAttribute.value.length !== 2 ) {
 					throw new Error( 'Value supplied for isBetween comparator must be an array [ min, max ]' );
 				}
 
-				if( thisObjectValue < thisKeyQuery.value[ 0 ] || thisObjectValue > thisKeyQuery.value[ 1 ] ) return false;
+				if( objectAttribute < queryAttribute.value[ 0 ] || objectAttribute > queryAttribute.value[ 1 ] ) return false;
+				break;
+			case 'startsWith':
+				if( ! objectAttribute ) return false;
+				// eslint-disable-next-line max-len
+				if( ! _.isString( objectAttribute ) || ! _.isString( queryAttribute.value ) ) throw new Error( 'For startsWith comparator both the object attribute and the query value must be strings.' );
+				if( ! _startsWith( objectAttribute, queryAttribute.value ) ) return false;
+				break;
+			case 'endsWith':
+				if( ! objectAttribute ) return false;
+				// eslint-disable-next-line max-len
+				if( ! _.isString( objectAttribute ) || ! _.isString( queryAttribute.value ) ) throw new Error( 'For endsWith comparator both the object attribute and the query value must be strings.' );
+				if( ! _endsWith( objectAttribute, queryAttribute.value ) ) return false;
 				break;
 			default:
-				throw new Error ( 'Invalid comparator ' + thisKeyQuery.comparator );
+				throw new Error ( 'Invalid comparator ' + queryAttribute.comparator );
 			}
-		} else if( _.isArray( thisKeyQuery ) ) {
-			if( ! _.contains( thisKeyQuery, object[ thisKey ] ) ) return false;
+		} else if( _.isArray( queryAttribute ) ) {
+			if( ! _.contains( queryAttribute, objectAttribute ) ) return false;
 		} else {
-			if( thisKeyQuery !== object[ thisKey ] ) return false;
+			if( queryAttribute !== objectAttribute ) return false;
 		}
 	}
-
 	return true;
 };
+
+function _startsWith( str, starts ) {
+	starts = String( starts );
```

Comment:
@martinflory we are already checking that this is a string... why cast it here?

Reply:
Yes! You're right. thanks